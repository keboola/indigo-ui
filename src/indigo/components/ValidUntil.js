import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import date from './date';

class ValidUntil extends React.Component {
  render() {
    if (this.props.validUntil) {
      return <span>N/A</span>;
    }
    return (
      <span title={date.format(this.props.validUntil)}>
        {this.props.showIcon && <i className="fa fa-clock-o duration-icon" />}
        {this.props.validUntil < moment.now()
          ? moment(this.props.validUntil).fromNow()
          : 'any time now'}
      </span>
    );
  }
}

ValidUntil.propTypes = {
  validUntil: PropTypes.number,
  showIcon: PropTypes.bool,
};

ValidUntil.defaultProps = {
  showIcon: false,
};

export default ValidUntil;
