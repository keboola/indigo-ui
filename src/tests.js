import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
