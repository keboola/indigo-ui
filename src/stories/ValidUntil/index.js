import React from 'react';
import {storiesOf} from '@storybook/react';
import {withInfo} from '@storybook/addon-info';
import ValidUntil from '../../indigo/components/ValidUntil';
import { boolean } from '@storybook/addon-knobs';

const now = new Date();
const timePast = now.getTime() - 50000000;
const timeFuture = now.getTime() + 50000000;

storiesOf('ValidUntil', module)
  .add(
    'ValidUntil',
    withInfo({
      inline: true,
    })(() => {
      return <ValidUntil validUntil={timePast} showIcon={boolean('showIcon', true)}/>;
    })
  )
  .add(
    'ValidUntil - valid any time',
    withInfo({
      inline: true,
    })(() => {
      return <ValidUntil validUntil={timeFuture} showIcon={boolean('showIcon', true)}/>;
    })
  )
  .add(
    'ValidUntil - empty state',
    withInfo({
      inline: true,
    })(() => {
      return <ValidUntil showIcon={boolean('showIcon', true)} />;
    })
  );
