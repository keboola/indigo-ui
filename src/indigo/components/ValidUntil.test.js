import React from 'react';
import moment from 'moment';
import {snapshot} from '../../tests';
import ValidUntil from './ValidUntil';

const fakeTimeNow = moment('2018-11-26T15:11:05+0000').valueOf();
const timePast = moment('2018-11-25T15:11:09+0000').valueOf();
const timeFuture = moment('2018-11-29T15:09:09+0000').valueOf();

global.Date.now = jest.fn(() => fakeTimeNow);

describe('<ValidUntil />', () => {
  it("should render 'a day ago'", () => {
    snapshot(<ValidUntil validUntil={timePast} />);
  });
  it("should render 'any time now'", () => {
    snapshot(<ValidUntil validUntil={timeFuture} />);
  });
  it("should render 'N/A'", () => {
    snapshot(<ValidUntil />);
  });
});
