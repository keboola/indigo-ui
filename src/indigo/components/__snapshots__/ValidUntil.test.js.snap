// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidUntil /> should render 'N/A' 1`] = `
<span>
  N/A
</span>
`;

exports[`<ValidUntil /> should render 'a day ago' 1`] = `
<span
  title="2018-11-25 16:11:09"
>
  a day ago
</span>
`;

exports[`<ValidUntil /> should render 'any time now' 1`] = `
<span
  title="2018-11-29 16:11:09"
>
  any time now
</span>
`;
